azure/scope/machine.go

```diff
@@ -491,7 +494,7 @@ func (m *MachineScope) AvailabilitySetSpec() azure.ResourceSpecGetter {
 
 // AvailabilitySet returns the availability set for this machine if available.
 func (m *MachineScope) AvailabilitySet() (string, bool) {
-	if !m.AvailabilitySetEnabled() {
+	if !m.AvailabilitySetEnabled() || m.ExtendedLocation() != nil {
```

Comment:
can you please add a short code comment about why `ExtendedLocation` disabled AvailabilitySets for future reference?

Reply:
Sure.